ghproxy/ghcache/ghcache.go

```diff
@@ -206,6 +193,16 @@ func (u upstreamTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	return resp, nil
 }
 
+func authHeaderHash(req *http.Request) string {
+	// get authorization header to convert to sha256
+	authHeader := req.Header.Get("Authorization")
+	if authHeader == "" {
+		logrus.Warn("Couldn't retrieve 'Authorization' header, adding to unknown bucket")
+		authHeader = "unknown"
+	}
+	return fmt.Sprintf("%x", sha256.Sum256([]byte(authHeader))) // use %x to make this a utf-8 string for use as a label
```

Comment:
Can we choose a computationally less expensive hash?

Reply:
This does need to be a cryptographic hash. Is SHA256 considered slow?
We could also keep a simple cache of hash values.

Reply:
Happy to follow up with a simple in-memory cache

Reply:
I do not agree that this needs to be a cryptographic hash. It needs to be a one-way function that has a low chance of collisions. If we intend to perform it on every request, it must be computationally cheap.

Reply:
Although it looks like we are already using sha256? In which case this isn't a regression so is less important to fix (but should be fixed, sha256 is almost certainly a poorer choice for this than say md5 given the need)

Reply:
I think the performance hit of this is miniscule but we should see when this lands ... I have added a simple in-memory cache. Isn't md5 cryptographic (but broken?)

Reply:
There are scenarios where md5 is unsuitable, but it isn't obvious to me how this is one of them